POFile support for sync_projects

Osmose opened the pull request:
This PR adds support for the po/pot formats to sync_projects. It also updates sync_projects to support plural strings and sources for formats that require them.

Because I had to modify sync_projects a bit, I based this PR off the existing #105 PR. That should be merged before this, and I can rebase once it lands. In the meantime, you can view individual commits to see the three extra commits this adds.

This also adds tests for the pofile parser. They're designed to be generic enough that any format can use them as long as they provide the proper input and output files for each test, but I haven't yet actually done the work yet to make them generic. I figure the PR for xliff can support that so we can have a common set of "requirements" that each format supports, with opt-ins to optional features (for example, testing for plurals on po and xliff formats, but not for langfiles).

I've also done local testing and things seem to work, using the Affiliates project as a testcase, but this could definitely use some more attention on a manual testing side. :D
@mathjazz r?

mathjazz commented:
Among other things, we should use this as a real-world test:
https://bugzilla.mozilla.org/show_bug.cgi?id=1193953

mathjazz commented:
r+, the code looks good. But I'd also like to test it on staging first.

You should probably unify classnames of POResource and LangFile (I see why you didn't use POFile).

Osmose commented:
Rebased now that the tests have been merged.

mathjazz commented on pontoon/base/tests/formats/test_po.py:
```diff
@@ -0,0 +1,385 @@
+        """)))
+
+    def test_save_plural_remove(self):
+        """Any missing plurals should be removed from the pofile."""
```
Not like this. You need to keep the length of msgstr to be nplurals, otherwise PO file will not be valid. When deleting strings, you basically need to set them to empty string: "".

Osmose commented:
I've addressed the feedback and this is ready to merge, but you mentioned wanting to test on staging, so I deployed this on staging. Lemme know when you're happy with this and I'll merge!
